lms/djangoapps/discussion/static/discussion/js/discussion_router.js

```diff
@@ -21,7 +20,7 @@
                     _.bindAll(this, 'allThreads', 'showThread');
                     this.courseId = options.courseId;
                     this.discussion = options.discussion;
-                    this.course_settings = new DiscussionCourseSettings(options.course_settings);
+                    this.course_settings = options.courseSettings;
```

Comment:
Can you rename this to ``courseSettings`` to reduce the risk of such mistakes again.